pkg/util/validation.go

```diff
@@ -18,19 +18,34 @@ package util
 
 import (
 	"regexp"
-	"strings"
 )
 
-// IsDNSLabel tests for a string that conforms to the definition of a label in
-// DNS (RFC 1123).
-func IsDNSLabel(value string) bool {
-	return IsDNS1123Label(value)
+const LabelValueFmt string = "[A-Za-z0-9_\\-\\.,]*"
+
+var labelValueRegexp = regexp.MustCompile("^" + LabelValueFmt + "$")
+
+const LabelValueMaxLength int = 63
+
+func IsValidLabelValue(value string) bool {
+	return (len(value) <= LabelValueMaxLength && labelValueRegexp.MatchString(value))
 }
 
-// IsDNSSubdomain tests for a string that conforms to the definition of a
-// subdomain in DNS (RFC 1123).
-func IsDNSSubdomain(value string) bool {
-	return IsDNS1123Subdomain(value)
+const AnnotationValueFmt string = ".*"
+
+var annotationValueRegexp = regexp.MustCompile("^" + AnnotationValueFmt + "$")
+
+const TotalAnnotationSizeB int = 64 * (1 << 10) // 64 kB
+
+func IsValidAnnotationValue(value string) bool {
+	return annotationValueRegexp.MatchString(value)
+}
+
+const QualifiedNameFmt string = "([A-Za-z0-9_\\-\\.]+/)?[A-Za-z0-9_\\-\\.]*[A-Za-z0-9]"
```

Comment:
Can we simplify this the same way we did DNS1123SubdomainFmt?  Also it is inconsistent.

First: we should decide whether we want to allow tokens that end in punctuation or not.

If yes, define KubeToken as: [-A-Za-z0-9_.]+
If no, define KubeToken as: [A-Za-z0-9]([-A-Za-z0-9_.]*[A-Za-z0-9])?

I don't really want to deal with fallout of things like "." being a valid key, so I'd go with the latter, probably.

Second, define QualifiedName as (KubeToken + "/")? + KubeToken

use KubeToken in validation for